File: src/my_class_pkg/scripts/ros_action_client.py
#!/usr/bin/env python3
import rospy
import actionlib
from my_class_pkg.msg import MyActionAction, MyActionGoal

def feedback_cb(feedback):
    rospy.loginfo('Progress: %s', feedback.progress)

if __name__ == '__main__':
    rospy.init_node('my_client')
    client = actionlib.SimpleActionClient('my_action', MyActionAction)
    client.wait_for_server()
    goal = MyActionGoal(object_name='world')
    rospy.loginfo('Sending goal...')
    client.send_goal(goal, feedback_cb=feedback_cb)
    client.wait_for_result()
    res = client.get_result()
    if res and res.success:
        rospy.loginfo('Action succeeded')
    else:
        rospy.loginfo('Action failed')

File: src/my_class_pkg/src/ros_action_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <my_class_pkg/MyActionAction.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "my_action_client");
    actionlib::SimpleActionClient<my_class_pkg::MyActionAction> client("my_action", true);
    client.waitForServer();
    ROS_INFO("Waiting for action server to start...");

    my_class_pkg::MyActionGoal goal;
    goal.object_name = "world";
    ROS_INFO("Sending goal...");
    client.sendGoal(goal);

    bool finished = client.waitForResult(ros::Duration(30.0));
    if (finished)
    {
        auto state = client.getState();
        ROS_INFO("Action finished: %s", state.toString().c_str());
    }
    else
    {
        ROS_INFO("Action did not finish before timeout.");
    }
    return 0;
}
